### PR TITLE
chore(payments): detach payment method on failed first sub payment

### PR DESCRIPTION
**Pull request by chenba**
Because:
 - a customer is required to enter a different, working, credit card
   after a failed payment during the subscription to their first product,
   so there's no reason to keep the payment method created by the failed
   payment

This commit:
 - have the Payments frontend call the auth-server to detach the payment
   method

## Issue that this pull request solves

Closes: #6336 (FXA-2494)

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.

### CODE REVIEW

**Comment by chenba** [comment]
Actually, I am not sure how _necessary_ this is.  It's a cleanup for an edge case.  (Functionality was fixed with PR #6353, so the customer is no longer dead-ended.)

But it's still good to keep the bad payment methods to a minimum. 🤷

File: packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx [hunk 1]
```diff
@@ -349,6 +361,16 @@ async function handlePaymentIntent({
...
     case 'requires_payment_method': {
+      // If this happens when a customer is trying to subscribe to their first
+      // product, detach the payment method.  The user must enter a working
+      // card to move on; we'll get a working payment method from that.
+      // This does not affect functionality, thus not blocking.
+      if (
+        (!customer || (customer && customer.subscriptions.length === 0)) &&
+        paymentMethodId
+      ) {
+        apiDetachFailedPaymentMethod({ paymentMethodId });
```
Comment (lmorchard) [inline packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx:372]: I think there might be a *slim* chance here that we could end up leaving a failed payment method in place - i.e. since we don't have retries, a network error at this point would do it. 

But it's so slim a chance that it's probably just a thing to keep in mind if a customer service issue comes up later.

**Review by lmorchard — APPROVED** [review]
LGTM, worst I could come up with is an edge condition that I *think* we already talked through in meetings
